### PR TITLE
alex/update ghc

### PR DESCRIPTION
**Pull request by alexbiehl**
- Allow building with recent GHCs
- Test recent GHCs during CI

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -5,19 +5,15 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
       - name: Setup Haskell
-        uses: haskell/actions/setup@v1
+        uses: haskell/actions/setup@v2.4.3
         with:
           cabal-version: latest
           ghc-version: ${{ matrix.ghc }}
-      - run: cabal v2-test
+      - run: cabal v2-build
```
Comment (phile314) [inline .github/workflows/build.yaml:19]: Doesn't this mean the tests are no longer run at all? I mean it's debatable if running the tests should be a separate step, but they should definitely be run at some point.

**Review by phile314 — CHANGES_REQUESTED** [review]
Mostly LGTM.

**Comment by LaurentRDC** [comment]
@alexbiehl Gentle ping to address the changes requested above. Your PR would allow users to use `pinch-gen` with GHC 9.8!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (alexbiehl) [reply .github/workflows/build.yaml:19]: Not sure why I changed this to be build only 🤔 maybe it’d fail calling cabal test on a package that doesn’t have tests.

**Comment by phile314** [comment]
Merged and released as 0.4.4.0.

**Comment by LaurentRDC** [comment]
@phile314 Thank you for making a release!

Would you be able to update the Hackage metadata to allow base-4.19 (GHC 9.8)? I can already use pinch-gen with GHC 9.8 using the following clause in a `cabal.project` file:

```
allow-newer: pinch-gen:base
```

**Comment by phile314** [comment]
I took out the sledge hammer and increased the base upper limit to 5 :smile: 

Let's see how this goes for the next few GHC releases.

**Comment by LaurentRDC** [comment]
I greatly appreciate this, thank you!

**Pull request merged**